gen3/bin/gitops.sh

```diff
@@ -123,35 +124,63 @@ gen3_gitops_sync() {
 
   # portal directory config check
   commonsManifestDir=$(dirname $(g3k_manifest_path))
+  local defaultsDir="${GEN3_HOME}/kube/services/portal/defaults"
   if [[ ! -d "${commonsManifestDir}/portal" ]]; then
     gen3_log_info "Portal directory not found, skipping portal update"
   else
-    # for each file in the portal dir, look for a diff in the portal-config secret
+    # for each file in the portal defaults dir...
+    #   if file not found in commons's portal dir
+    #     use default file for comparison
+    #   else
+    #     use commons's file in manifest/portal dir
+    #
+    #   if file not in secret
+    #     roll portal
+    #   else if comparison file and secret are different
+    #     roll portal
+
     local filename
-    local oldFile
-    local newFile
-    for filepath in $(find "${commonsManifestDir}/portal/" -type f); do
-      filename=$(basename ${filepath})
-      newFile=$(base64 $filepath)
-      oldFile=$(g3kubectl get secret portal-config -o json | jq -r '.data."'"${filename}"'"')
-      # found diff if they aren't equal or if there's a new file not found in the secret
-      if [[ -z "${oldFile// }" ]]; then
-        gen3_log_info "Found portal diff in portal/${filename} - filename not found in secret or is empty"
-        portal_roll=true
-      elif [[ "${oldFile}" != "${newFile}" ]]; then
-        gen3_log_info "Found diff in portal/${filename} - difference between file and secret"
+    local secretsFile
+    local comparingFile
+    local diffMsg
+    for defaultFilepath in $(find "${defaultsDir}/" -name "gitops*" -type f); do
+      diffMsg=""
+      filename=$(basename ${defaultFilepath})
+      commonsFilepath="${commonsManifestDir}/portal/$filename"
+      secretsFile=$(g3kubectl get secret portal-config -o json | jq -r '.data."'"${filename}"'"')
```

Comment:
do `.data["$filename"]` - data.bla breaks if bla has unexpected characters in it

Reply:
I have it wrapped in " ", which does the same thing as [ ]